File: result/repair/transcoder/java2python/python_repair/SUBARRAYS_DISTINCT_ELEMENTS.py
def f_gold ( arr , n ) :
    s = [ ]
    j = 0
    ans = 0
    for i in range ( n ) :
        while j < n and not s.count ( arr [ j ] ) :
            s.append ( arr [ j ] )
            j += 1
        ans += ( ( j - i ) * ( j - i + 1 ) ) // 2
        s.remove ( int ( arr [ i ] ) )
    return ans

File: result/translate/chatgpt/java2cpp/c++/SUBARRAYS_DISTINCT_ELEMENTS.cpp
int f_gold(int arr[], int n) {
    vector<int> s;
    int j = 0;
    int ans = 0;
    for (int i = 0; i < n; i++) {
        while (j < n && find(s.begin(), s.end(), arr[j]) == s.end()) {
            s.push_back(arr[j]);
            j++;
        }
        ans += ((j - i) * (j - i + 1)) / 2;
        s.erase(remove(s.begin(), s.end(), arr[i]), s.end());
    }
    return ans;
}